### PR TITLE
Make gulp:all and CI-related tasks work

### PR DESCRIPTION
**Pull request by surma**
`jscs` wants JSDoc comments on every function in the gulpfile and makes drone fail. I’m arguing we don’t need to lint our gulpfile.

r: @sgomes @addyosmani cc: @crhym3

### CODE REVIEW

**Comment by crhym3** [comment]
Here's a sample of lint errors: https://ci.cloudware.io/google/material-design-lite/3

**Comment by crhym3** [comment]
@surma can you rebase this pr to the latest master? I've just pushed correct drone config.

**Comment by samccone** [comment]
@surma what about just putting a jsdoc ignore at the top of this file?

I think it is a good idea to still lint this file :8ball:

**Comment by surma** [comment]
@samccone LOL. Forgot about that option. Much better, thanks for reminding me. Done.

@crhym3 Done.

**Comment by crhym3** [comment]
@surma mind rebasing this to master again? I think I fixed phantomjs weirdness in https://github.com/google/material-design-lite/commit/b1f840d3b6690540d6d123365b1c34bed2198aea

**Comment by surma** [comment]
@crhym3 done

**Comment by crhym3** [comment]
damnit. still failing on phantom :(

**Comment by crhym3** [comment]
https://github.com/google/material-design-lite/commit/e03f7f9a4fabf62cdaf541f53884d461a311e47e is what's apparently needed to make gulp tasks run within CI context.

However, test/memory/test.js still fails with:

```
$ ./node_modules/.bin/babel-node test/memory/test.js
undefined:1190
   vlog(2, () => this + ' scheduling notifications', this);
            ^
SyntaxError: Unexpected token )
```

I guess this PR + https://github.com/google/material-design-lite/commit/e03f7f9a4fabf62cdaf541f53884d461a311e47e + a fix for test/memory/test.js is what will ultimately make the CI work.

**Comment by samccone** [comment]
lolololol

[image](https://twitter.com/samccone/status/671443757125017600)

**Comment by crhym3** [comment]
Right. And while someone's making those changes, @samccone also suggested https://github.com/google/material-design-lite/commit/b1f840d3b6690540d6d123365b1c34bed2198aea#commitcomment-14886951 which is as easy as modifying `CHROME` and/or `WEBDRIVER` in .drone.yml.

**Comment by sgomes** [comment]
The version of `drool` we're using isn't correctly requesting the version of `selenium-webdriver` it needs, and ends up using one that relies on ES6. We should update `package.json` to request `"drool": "^0.3.1"` instead (and then solve the test failures that seem to be happening).

**Comment by crhym3** [comment]
Good catch. 
So, I've just tried with drool ^0.3.1 and the memory test now runs indeed but fails with:

```
/drone/src/github.com/google/material-design-lite/node_modules/selenium-webdriver/lib/goog/async/nexttick.js:39
  goog.global.setTimeout(function() { throw exception; }, 0);
                                            ^
AssertionError: true == false
    at /drone/src/github.com/google/material-design-lite/test/memory/test.js:46:12
```

Seems like it can't compute the numbers anymore:

```
$ node test/memory/test.js
**MaterialButton** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
NaN|0.00 PB|NaN

**MaterialSpinner** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
NaN|0.00 PB|NaN

...
```

You can see the full log here:
https://ci.cloudware.io/google/material-design-lite/19

**Comment by sgomes** [comment]
@samccone Any idea what might be happening there?

**Comment by samccone** [comment]
in 0.3.x of drool I changed the API (yay semver)
https://github.com/samccone/drool/commit/4f2e5b5f8aed559210295d5ed298a77c421845e5

You will want to update these lines as follows

https://github.com/google/material-design-lite/blob/master/test/memory/menu.js#L21

```diff
-  stamps.push([after.jsEventListeners, initial.jsEventListeners]);
+  stamps.push([after.counts.jsEventListeners, initial.counts.jsEventListeners]);
```

https://github.com/google/material-design-lite/blob/master/test/memory/snackbar.js#L19

```diff
-  stamps.push([after.jsEventListeners, initial.jsEventListeners]);
+  stamps.push([after.counts.jsEventListeners, initial.counts.jsEventListeners]);
```


:+1:

**Comment by surma** [comment]
Can you feel the pain of the SemVer whip on your back, @samccone?

Added the drool upgrade and the fix to this PR.

**Comment by samccone** [comment]
back to phantom errors :ghost:

File: gulpfile.babel.js [hunk 1]
```diff
@@ -17,6 +17,8 @@
  *
  */
 
+ // jscs:disable jsDoc
```
Comment (Garbee) [inline gulpfile.babel.js]: Can we please get this just tossed directly into `master` ASAP? It is a major headace adding it in on all these branches to test things out repeatedly. The rest of this PR seems like extra semi-related fixes but not needed for the direct `gulp all` fails as far as I can tell.

**Comment by crhym3** [comment]
Give me a second. I'll merge my changes in drone branch into this PR and it'll all should work.

**Comment by crhym3** [comment]
@sgomes @surma @Garbee it seems like we also have some closure errors?

```
[11:02:48] Starting 'closure'...

stream.js:94
      throw er; // Unhandled stream error in pipe.
            ^
Error: src/layout/layout.js:161: WARNING - Property toggleDrawer_ never defined on MaterialLayout
      this.toggleDrawer_();
           ^

src/layout/layout.js:200: WARNING - Property toggleDrawer_ never defined on MaterialLayout
    this.toggleDrawer_();
         ^

src/layout/layout.js:208: WARNING - variable MaterialLayout.prototype.drawerToggleHandler_ redefined with type function (this:MaterialLayout, (Event|null)=): undefined, original definition at src/layout/layout.js:189 with type function (this:MaterialLayout, (Event|null)): undefined
  MaterialLayout.prototype.drawerToggleHandler_ = function() {
  ^
```

**Comment by crhym3** [comment]
nvm, it's in the master branch. should be a separate fix.

**Comment by sgomes** [comment]
@crhym3 Do you mean it's already fixed in master? If not, I can take a look.

**Comment by crhym3** [comment]
@sgomes no, sorry. I meant those closure errors seems to be appeared in the master while we've been fixing gulp and drone. Probably because drone isn't running properly, some PRs got merged w/o gulp mocha:closure check.

I suspect if you run `gulp mocha:closure` from master now, you'll see those errors.

**Comment by crhym3** [comment]
btw, @samccone, memory test isn't fixed:

```
$ node test/memory/test.js
**MaterialButton** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
NaN|0.00 PB|NaN

**MaterialSpinner** .. run: 1

node delta | heap delta | event listener
--- | --- | --- |
NaN|0.00 PB|NaN
```

see bottom of https://ci.cloudware.io/google/material-design-lite/32

**Comment by crhym3** [comment]
Actually, @samccone it's not an error, just incorrect output. I'm pretty sure it has something to do with [test/memory/utils.js](https://github.com/google/material-design-lite/blob/857fe1f401e506b607138cdda135b4bf1eb420e7/test/memory/utils.js).

**Comment by crhym3** [comment]
This last commit should fix it. Let's see...

**Comment by crhym3** [comment]
@sgomes I think this is now good to merge.

**Comment by crhym3** [comment]
@Garbee I added you to drone users, so you should be able to restart builds manually, even though I still  don't get why need it :)

**Comment by surma** [comment]
Thanks for taking care of this, @crhym3!

**Comment by sgomes** [comment]
Thanks, @crhym3! Feel free to merge at will, from my end. Once the CI is working again, I'll take a look at the closure issues and fix them.

**Comment by crhym3** [comment]
Just to confirm, here are closure errors after merging with master:

```
$ gulp mocha:closure
[13:52:27] Requiring external module babel-core/register
[13:52:28] Using gulpfile /drone/src/github.com/google/material-design-lite/gulpfile.babel.js
[13:52:28] Starting 'closure'...

stream.js:94
     throw er; // Unhandled stream error in pipe.
           ^
Error: src/layout/layout.js:161: WARNING - Property toggleDrawer_ never defined on MaterialLayout
     this.toggleDrawer_();
          ^

src/layout/layout.js:200: WARNING - Property toggleDrawer_ never defined on MaterialLayout
   this.toggleDrawer_();
        ^

src/layout/layout.js:208: WARNING - variable MaterialLayout.prototype.drawerToggleHandler_ redefined with type function (this:MaterialLayout, (Event|null)=): undefined, original definition at src/layout/layout.js:189 with type function (this:MaterialLayout, (Event|null)): undefined
 MaterialLayout.prototype.drawerToggleHandler_ = function() {
 ^
```

See bottom of https://ci.cloudware.io/google/material-design-lite/36
This will prevent drone from auto-updating mdl-staging.

**Comment by sgomes** [comment]
Thanks, @chrym3, I'm on it :)

**Comment by Garbee** [comment]
Those layout things are my fault from a bad layout merge yesterday. Git didn't show me the full conflict area it seems. Fixing it right now.

**Comment by crhym3** [comment]
@Garbee it's already been fixed by @sgomes I believe.

**Comment by crhym3** [comment]
@Garbee https://github.com/google/material-design-lite/commit/a2ae0208a1f87b8ad27e46049351b9062aa3ad0a

**Comment by Garbee** [comment]
Yup, just found that. Thanks guys!

**Pull request merged**